File: frontend/src/services/departmentService.js
import axiosClient from "../api/axiosClient";

export const getDepartments = async (
    page = 0,
    size = 10
) => {

    const response =
        await axiosClient.get(
            `/departments?page=${page}&size=${size}`
        );

    return response.data;
};

export const createDepartment =
    async (departmentData) => {

        const response =
            await axiosClient.post(
                "/departments",
                departmentData
            );

        return response.data;
    };

export const updateDepartment =
    async (
        id,
        departmentData
    ) => {

        const response =
            await axiosClient.put(
                `/departments/${id}`,
                departmentData
            );

        return response.data;
    };

export const deactivateDepartment =
    async (id) => {

        const response =
            await axiosClient.patch(
                `/departments/${id}/deactivate`
            );

        return response.data;
    };

export const activateDepartment =
    async (id) => {

        const response =
            await axiosClient.patch(
                `/departments/${id}/activate`
            );

        return response.data;
    };

File: backend/nextalx-api/src/main/java/com/nextalx/service/DepartmentService.java
package com.nextalx.service;

import com.nextalx.dto.request.CreateDepartmentRequest;
import com.nextalx.dto.request.UpdateDepartmentRequest;
import com.nextalx.dto.response.DepartmentResponse;
import org.springframework.data.domain.Page;

public interface DepartmentService {

    Page<DepartmentResponse> getAllDepartments(
            int page,
            int size
    );

    DepartmentResponse createDepartment(
            CreateDepartmentRequest request
    );

    DepartmentResponse updateDepartment(
            Long id,
            UpdateDepartmentRequest request
    );

    DepartmentResponse deactivateDepartment(
            Long id
    );

    DepartmentResponse activateDepartment(
            Long id
    );
}